s3fs/tests/test_s3fs.py

```diff
@@ -256,6 +256,38 @@ def test_ls_touch(s3):
     assert set(L) == {a, b}
 
 
+def test_isfile(s3):
+    assert s3.isfile(test_bucket_name+'/test/accounts.1.json')
+    assert s3.isfile(test_bucket_name+'/test/accounts.2.json')
+    assert not s3.isfile(test_bucket_name + '/test/accounts.3.json')
+    assert not s3.isfile(test_bucket_name + '/test')
+    assert not s3.isfile(test_bucket_name)
+
+    assert not s3.isfile(a)
+    s3.touch(a)
+    assert s3.isfile(a)
+
+    assert not s3.isfile(b)
+    s3.mkdir(b)
+    assert not s3.isfile(b)
```

Comment:
I am surprised that this assert passes - mkdir works by calling touch. Perhaps isfile only fails because the path that is passed to touch has '/' appended, i.e., `s3.isfile(b + '/') and s3.isdir(b + '/') is True`.

Reply:
Good catch, fixed and added more tests at c442c5d17edde54e8a3ea1232f5577ff8dad28e7